File: MongoDB.js
const MongoClient = require('mongodb').MongoClient;
const uri = "mongodb+srv://bigdatamanagement:Bigdatafresh2020%3F@mongosandbox-qzmsu.mongodb.net/test?retryWrites=true&w=majority";
const client = new MongoClient(uri, { useNewUrlParser: true });
client.connect(err => {
    const collection = client.db("projectfour").collection("recipes");
    // perform actions on the collection object
    var query = {preparation_time: "35 minutes"};
    collection.find(query).toArray(function(err, result) {
        if (err) throw err;
        console.log(result)
    });
    client.close();
});

// Start connection to mongodb

File: lambda-python/MongoDB.py
from pymongo import MongoClient


def get_recipes_db():
    client = MongoClient(
        "mongodb+srv://bigdatamanagement:bigdatamanagement2019@mongosandbox-qzmsu.mongodb.net/test?retryWrites=true&w=majority")
    db = client['projectfour']
    recipes_db = db.get_collection('recipes')
    return recipes_db, client


# def mongo_to_csv():
#     recipes_db, client = get_recipes_db()
#     recipes = recipes_db.find()
#     factors = list()
#     for mongo_recipe in recipes:
#         recipe_factors = list()
#         recipe = HelloFreshRecipe.from_mongo(mongo_recipe)
#         factors.append(recipe.to_factors())
#     print(factors)
#     write_list_csv("data", "recipe_stats_ratings.csv", factors)


